src/Sextant.UWP/NavigationView.xaml.cs

```diff
@@ -0,0 +1,456 @@
+                .FromEvent<PointerEventHandler, PointerRoutedEventArgs>(
+                handler =>
+                {
+                    PointerEventHandler pointerHandler = (sender, e) => handler(e);
+                    return pointerHandler;
+                },
+                x => PointerPressed += x,
+                x => PointerPressed -= x)
+                .Where(args => args.GetCurrentPoint(this as UIElement).Properties.PointerUpdateKind == Windows.UI.Input.PointerUpdateKind.XButton1Pressed)
+                .Do(args =>
+                {
+                    if (mainFrame.CanGoBack)
+                    {
+                        PopPage(true);
+                        args.Handled = true;
+                    }
+                })
+                .ToSignal(),
+            Observable
+                .FromEvent<RoutedEventHandler, RoutedEventArgs>(
+                handler =>
+                {
+                    RoutedEventHandler routedHandler = (sender, e) => handler(e);
+                    return routedHandler;
+                },
+                x => backButton.Click += x,
+                x => backButton.Click -= x)
+                .Do(args =>
+                {
+                    if (mainFrame.CanGoBack)
+                    {
+                        PopPage(true);
+                    }
+                })
+                .ToSignal());
+
+        /// <inheritdoc />
+        public IObservable<Unit> PopModal()
+        {
+            if (_contentDialog != null)
+            {
+                _contentDialog.Hide();
+                var viewStackService = Locator.Current.GetService<IViewStackService>();
+
+                viewStackService.ModalStack.Skip(1).Take(1).Subscribe(stack =>
+                {
+                    if (stack.Count > 0)
+                    {
+                        _contentDialog = new ContentDialog();
+                        _contentDialog.FullSizeDesired = true;
+                        _contentDialog.IsPrimaryButtonEnabled = false;
+                        _contentDialog.IsSecondaryButtonEnabled = false;
+                        var page = LocatePageFor(stack[stack.Count - 1], null);
+
+                        _contentDialog.Content = page;
+
+                        _ = _contentDialog.ShowAsync();
+                    }
+                });
+            }
+
+            return Observable.Return(Unit.Default).ObserveOn(_mainScheduler);
+        }
+
+        /// <inheritdoc />
+        public IObservable<Unit> PopPage(bool animate)
+        {
+            NavigationTransitionInfo animation;
+            if (animate)
+            {
+                animation = new Windows.UI.Xaml.Media.Animation.EntranceNavigationTransitionInfo();
+            }
+            else
+            {
+                animation = new Windows.UI.Xaml.Media.Animation.SuppressNavigationTransitionInfo();
+            }
+
+            _currentViewModel = (IViewModel)(mainFrame.Content as IViewFor).ViewModel;
```

Comment:
I wasn't sure if a private getter was kosher so it's a method now.

Reply:
Actually, no... there's no way to get the ViewModel that was just popped without storing it temporarily somewhere.  UWP's `Navigated` event only provides the new view... and it might not be the original one even if it's a back navigation.  Caching views is a developer opt-in for specific pages.  

This was the best method I could think of without changing Sextant or writing my own `Frame` with navigation.

Reply:
and now I realize it should have been in `PopToRootPage` as well.

Reply:
I changed a lot... will comment in main